src/Microsoft.SqlTools.ServiceLayer/EditData/EditSession.cs

```diff
@@ -516,12 +516,24 @@ private async Task CommitEditsInternal(DbConnection connection, Func<Task> succe
                 editOperations.Sort();
                 foreach (var editOperation in editOperations)
                 {
-                    // Get the command from the edit operation and execute it
-                    using (DbCommand editCommand = editOperation.GetCommand(connection))
-                    using (DbDataReader reader = await editCommand.ExecuteReaderAsync())
+                    try
                     {
-                        // Apply the changes of the command to the result set
-                        await editOperation.ApplyChanges(reader);
+                        // Get the command from the edit operation and execute it
+                        using (DbCommand editCommand = editOperation.GetCommand(connection))
+                        using (DbDataReader reader = await editCommand.ExecuteReaderAsync())
+                        {
+                            // Apply the changes of the command to the result set
+                            await editOperation.ApplyChanges(reader);
+                        }
+                    }
+                    catch (Exception e)
+                    {
+                        if (e.ToString().Contains("delete"))
```

Comment:
can you create a new Exception-derived type and catch just that.  This isn't the correct pattern for detecting a particular type of exception.